src/utils/txHelpers.js

```diff
@@ -158,10 +164,13 @@ const argBuilderCallback = Object.freeze({
     ];
   },
   proposeActionSafe({ values, formData }) {
-    const hexData = safeEncodeHexFunction(
-      JSON.parse(values.abiInput),
-      values?.abiArgs || [],
-    );
+    const hexData =
```

Comment:
It all checks out. The only thing I'd say is that we're doing the same thing three times over. Can we refactor this into a function? Something with a nice semantic name like ensureHex() or whatever?